pkg/orchestrator/data_scrubber.go

```diff
@@ -0,0 +1,127 @@
+}
+func (ds *DataScrubber) ContainsBlacklistedWord(word string) bool {
+	for _, pattern := range ds.LiteralSensitivePatterns {
+		if strings.Contains(strings.ToLower(word), pattern) {
+			return true
+		}
+	}
+	return false
+}
+
+// ScrubCommand hides the argument value for any key which matches a "sensitive word" pattern.
+// It returns the updated cmdline, as well as a boolean representing whether it was scrubbed
+// future: we can add a check to do regex matching or simple matching depending whether we have RegexSensitivePatterns
+func (ds *DataScrubber) ScrubSimpleCommand(cmdline []string) ([]string, bool) {
+	changed := false
+	regexChanged := false
+	var wordReplacesIndexes []int
+
+	rawCmdline := strings.Join(cmdline, " ")
+	for _, pattern := range ds.RegexSensitivePatterns {
+		if pattern.MatchString(rawCmdline) {
+			regexChanged = true
+			rawCmdline = pattern.ReplaceAllString(rawCmdline, "${key}${delimiter}********")
+		}
+	}
+	newCmdline := strings.Split(rawCmdline, " ")
+	// preprocess, without the preprocessing it is needed to strip until the whitespaces.
+	for index, cmd := range newCmdline {
+		for _, pattern := range ds.LiteralSensitivePatterns { // this can be optimized
+			// if we found a word from the list, it means that either the current or next word should be a password we want to replace.
+			if strings.Contains(strings.ToLower(cmd), pattern) { // password=1234
+				if index == 0 {
+					// the first index | should be the command name which shouldn't be matched
+					continue
+				}
+				changed = true
+				v := strings.IndexAny(cmd, "=:")
+				if v > 1 {
+					// password:1234  password=1234 ==> password=****** || password:******
+					newCmdline[index] = cmd[:v+1] + "********"
+					// replace from v to end of string with ********
+					break
+				} else {
+					wordReplacesIndexes = append(wordReplacesIndexes, index+1)
+					break
+				}
+			}
+		}
+	}
+	for i := 0; i < len(wordReplacesIndexes); i++ {
+		// we still want to make sure that we are in the index e.g. the word is at the end and actually does not mean adding a password/token.
+		index := wordReplacesIndexes[i]
+		if index < len(newCmdline) {
+			if newCmdline != nil {
+				// we only want to replace words
+				for newCmdline[index] == "" {
+					index++
+				}
+				if index < len(newCmdline) {
+					newCmdline[index] = "********"
+				}
+			}
+		}
+	}
+
+	// if nothing changed, just return the input
+	if !(changed || regexChanged) {
+		return cmdline, false
+	}
+
+	return newCmdline, changed || regexChanged
+}
+
+// AddCustomSensitiveWords adds custom sensitive words on the DataScrubber object
+// In the future we can add own regex expression
+func (ds *DataScrubber) AddCustomSensitiveWords(words []string) {
+	ds.LiteralSensitivePatterns = append(ds.LiteralSensitivePatterns, words...)
+}
+
+// AddCustomSensitiveRegex adds custom sensitive regex on the DataScrubber object
+func (ds *DataScrubber) AddCustomSensitiveRegex(words []string) {
```

Comment:
I have no problem removing this at all. Initially I added this for cases where we want special handling of special input. I tried to make both (past) and current implementation work together. 👍

Reply:
Gotcha, let's make sure users can't add their own regex right now otherwise we're guaranteed to get GH issues with people shooting themselves in the foot because the cpu usage went up surprising quickly due to a misguided regex. But having the feature here for us to add features later is 👍

Reply:
I added the option for users to add their own words:
```
	if k := key(orchestratorNS, "custom_sensitive_words"); config.Datadog.IsSet(k) {
		oc.Scrubber.AddCustomSensitiveWords(config.Datadog.GetStringSlice(k))
	}
```

There is no option for custom regex 👍